core/src/main/java/org/infinispan/interceptors/xsite/NonTransactionalBackupInterceptor.java

```diff
@@ -144,15 +144,19 @@ private Object handleMultipleKeysWriteReturn(InvocationContext ctx, WriteCommand
       LocalizedCacheTopology localizedCacheTopology = clusteringDependentLogic.getCacheTopology();
       for (Object key : writeCommand.getAffectedKeys()) {
          DistributionInfo info = localizedCacheTopology.getDistribution(key);
-         if (info.isWriteOwner()) { //all owners need to keep track.
-            iracManager.trackUpdatedKey(info.segmentId(), key, writeCommand.getCommandInvocationId());
-         }
          if (!info.isPrimary()) {
             if (log.isTraceEnabled()) {
                log.tracef("Not replicating write to key %s as the primary owner is %s", key, info.primary());
             }
+            if (!ctx.isOriginLocal() && info.isWriteOwner()) {
+               // backup owner in remote context: needs to track the key
+               iracManager.trackUpdatedKey(info.segmentId(), key, writeCommand.getCommandInvocationId());
```

Comment:
IMO it would be clearer if `trackUpdatedKey()` would be called in one place, before `if (!info.isPrimary())`, with a comment explaining that keys need to be tracked (I assume) while holding the key lock.

Reply:
It can't! 
If the backup owner is the originator, this method is invoked twice and we want to invoke `trackUpdatedKey()` only in the remote context.
If the non-owner is the originator, this method in invoked once and we do not want to invoke `trackUpdatedKey()`

Reply:
I was thinking of something like this

```
         if (info.isPrimary() || (!ctx.isOriginLocal() && info.isWriteOwner())) {
            iracManager.trackUpdatedKey(info.segmentId(), key, writeCommand.getCommandInvocationId());
         }
```

We have the same logic in other interceptors, because we don't want e.g. to commit an entry on the originator when it's a backup owner, we want to commit it while the primary is holding the key lock.